odl/solvers/util/steplen.py

```diff
@@ -118,16 +138,20 @@ def __call__(self, x, direction, dir_derivative):
             The current point
         direction : `LinearSpaceElement`
             Search direction in which the line search should be computed
-        dir_derivative : float
+        dir_derivative : float, optional
             Directional derivative along the ``direction``
+            Default: ``function.gradient(x).inner(direction)``
```

Comment:
This requires `function.gradient` to exist, i.e. `function` to be a functional. That's obvious here, but users usually don't see this docstring, so a hint in `__init__` at the explanation of `function` would be better.